tests/Domain/Integrations/Repositories/EloquentIntegrationRepositoryTest.php

```diff
@@ -698,6 +702,89 @@ public function test_it_can_not_save_integration_with_subscription_with_differen
         );
     }
 
+    /** @dataProvider dataProviderGetIntegrationsThatHaveNotBeenActivatedYet */
+    public function test_get_integrations_that_have_not_been_activated_yet(IntegrationType $integrationType, IntegrationStatus $status, Carbon $date, ?Carbon $reminderEmailSent, bool $hasContact, int $expectedCount): void
+    {
+        $integrationId = Uuid::uuid4()->toString();
+        DB::table('integrations')->insert([
+            'id' => $integrationId,
+            'type' => $integrationType->value,
+            'subscription_id' => Uuid::uuid4()->toString(),
+            'name' => 'Test',
+            'description' => 'test',
+            'status' => $status,
+            'created_at' => $date,
+            'reminder_email_sent' => $reminderEmailSent,
+        ]);
+
+        if ($hasContact) {
+            DB::table('contacts')->insert([
+                'id' => Uuid::uuid4()->toString(),
+                'integration_id' => $integrationId,
+                'email' => 'grote.smurf@example.com',
+                'type' => ContactType::Technical->value,
+                'first_name' => 'Grote',
+                'last_name' => 'Smurf',
+            ]);
+        }
+
+        $this->assertCount($expectedCount, $this->integrationRepository->getIntegrationsThatHaveNotBeenActivatedYetByType(IntegrationType::SearchApi, 12));
+    }
+
+    public static function dataProviderGetIntegrationsThatHaveNotBeenActivatedYet(): array
+    {
+        return [
+            'Should not be selected: wrong type' => [
+                IntegrationType::EntryApi,
+                IntegrationStatus::Active,
+                Carbon::now()->subYears(2),
```

Comment:
This will fail after a certain time, you can use `Carbon::setTestNow` or even more advanced testing features explained here https://laravel.com/docs/11.x/mocking#interacting-with-time

Reply:
I am familiar with the theory behind time in unit tests and Carbon, I use it in SendIntegrationActivationReminderEmailTest, but I do not think it is a problem here.

I create a fresh integration each time with the time of "now", and then I create a date X months/years ago, and then I directly check it with a query.
I do not need to directly assert this date, and the test does not work with a fixed date.

Am I overlooking something?

Reply:
You are correct, it will work in the future because a new time is generated for every provided test.